tree/treeplayer/inc/TTreeReaderArray.h

```diff
@@ -59,90 +59,159 @@ namespace Internal {
 } // namespace ROOT
 
 template <typename T>
-class TTreeReaderArray: public ROOT::Internal::TTreeReaderArrayBase {
+class TTreeReaderArray : public ROOT::Internal::TTreeReaderArrayBase {
 public:
+   /// Create an array reader of branch "branchname" for TTreeReader "tr".
+   TTreeReaderArray(TTreeReader &tr, const char *branchname)
+      : TTreeReaderArrayBase(&tr, branchname, TDictionary::GetDictionary(typeid(T))) {}
+
+   T &At(std::size_t idx) { return *static_cast<T *>(UntypedAt(idx)); }
+   const T &At(std::size_t idx) const { return *static_cast<T *>(UntypedAt(idx)); }
+   T &operator[](std::size_t idx) { return At(idx); }
+   const T &operator[](std::size_t idx) const { return At(idx); }
+
+   /// Random access iterator to the elements of a TTreeReaderArray
+   class Iterator_t : public std::iterator<std::random_access_iterator_tag, T> {
+      TTreeReaderArray *fArray; ///< The array iterated over; nullptr if invalid/past-the-end.
+      std::size_t fIndex;       ///< Current index in the array.
+      std::size_t fSize;        ///< Size of the TTreeReaderArray
```

Comment:
Do we know that the iterator is invalidated when a new entry is loaded? I guess that's fine, right? ('Cause before, the size was implicitly allowed to change.)

Reply:
uhm, what do you expect to happen if I get an iterator, possibly use it, then load a new entry and try to keep using it?
should we support this usage? should we not, and warn users that do it?